packages/core/src/scuttle.js

```diff
@@ -0,0 +1,185 @@
+}
+
+/**
+ * Runs scuttling on the globalRef. Use applyDefaultScuttling for full scope of options.
+ *
+ * @param {GlobalRef} globalRef - Reference to the global object.
+ * @param {Array<string|RegExp>} extraPropsToAvoid - List of additional properties to exclude from scuttling beyond the default ones.
+ */
+function performScuttleGlobalThis(globalRef, extraPropsToAvoid = new Array()) {
+  const props = new Array()
+  getPrototypeChain(globalRef).forEach((proto) =>
+    props.push(...getOwnPropertyNames(proto))
+  )
+
+  // support LM,SES exported APIs and polyfills
+  const avoidForLavaMoatCompatibility = ['Compartment', 'Error', 'globalThis']
+  const propsToAvoid = new Set([
+    ...avoidForLavaMoatCompatibility,
+    ...extraPropsToAvoid,
+  ])
+
+  const obj = create(null)
+  for (const prop of props) {
+    // eslint-disable-next-line no-inner-declarations
+    function set() {
+      console.warn(
+        `LavaMoat - property "${prop}" of globalThis cannot be set under scuttling mode. ` +
+          'To learn more visit https://github.com/LavaMoat/LavaMoat/pull/360.'
+      )
+    }
+    // eslint-disable-next-line no-inner-declarations
+    function get() {
+      throw new Error(
+        `LavaMoat - property "${prop}" of globalThis is inaccessible under scuttling mode. ` +
+          'To learn more visit https://github.com/LavaMoat/LavaMoat/pull/360.'
+      )
+    }
+    if (shouldAvoidProp(propsToAvoid, prop)) {
+      continue
+    }
+    let desc = getOwnPropertyDescriptor(globalRef, prop)
+    if (desc?.configurable === true) {
+      desc = { configurable: false, set, get }
+    } else if (desc?.writable === true) {
+      const p = new Proxy(obj, { getPrototypeOf: get, get, set })
+      desc = { configurable: false, writable: false, value: p }
+    } else {
+      continue
+    }
+    defineProperty(globalRef, prop, desc)
+  }
+}
+
+/**
+ * @param {Set<string|RegExp>} propsToAvoid - List of properties to exclude from scuttling.
+ * @param {string} prop - Property to check.
+ * @returns {boolean} - Whether the property should be avoided or not.
+ */
+function shouldAvoidProp(propsToAvoid, prop) {
+  for (const avoid of propsToAvoid) {
+    if (avoid instanceof RegExp && avoid.test(prop)) {
+      return true
+    }
+    if (propsToAvoid.has(prop)) {
+      return true
+    }
+  }
+  return false
+}
+
+/**
+ * @param {object} value - object to get the prototype chain from.
+ * @returns {Array<object>} - Prototype chain as an array.
+ */
+function getPrototypeChain(value) {
+  const protoChain = []
+  let current = value
+  while (
+    current &&
+    (typeof current === 'object' || typeof current === 'function')
+  ) {
+    protoChain.push(current)
+    current = Reflect.getPrototypeOf(current)
+  }
+  return protoChain
+}
+
+/**
+ * @type {object}
+ * @property {Function} performScuttleGlobalThis - Scuttle the globalThis object once.
+ * @property {Function} applyDefaultScuttling - Apply default scuttling from options.
+ */
```

Comment:
```suggestion
```

Use of `Function` is a) discouraged by TS and b) causes the information about the function's parameters and return value to be lost.  It's also an object literal, which will be inferred by TS.